sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosClientBuilder.java

```diff
@@ -490,6 +492,15 @@ public CosmosClientBuilder preferredRegions(List<String> preferredRegions) {
         return this;
     }
 
+    CosmosClientBuilder dataEncryptionKeyProvider(DataEncryptionKeyProvider dataEncryptionKeyProvider) {
```

Comment:
How are the soft contracts managed? 
Are there gates to treat them as contract and don't regress?